lib/msal-common/src/cache/entities/CredentialEntity.ts

```diff
@@ -31,6 +31,8 @@ export type CredentialEntity = {
     tokenType?: AuthenticationScheme;
     /** KeyId for PoP and SSH tokens stored in the kid claim */
     keyId?: string;
+    /** KeyId for PoP passed in by client app in token request */
+    externalKeyId?: string;
```

Comment:
Do we have to treat an external key differently from an internal one?

Reply:
We do, if the externalKeyId is present, it is used to match the access token in a silent request. This is because for the regular PoP flow, the keyId is not part of the silent request, it only uses the AT attributes to match and then extracts the keyId from the cnf claim and checks to see if the key is in IndexedDB. If it's not, it breaks.

Reply:
Yes. This is mainly used for silent token refreshes. In the case where we already have a PoP Token in the cache with same scopes, and the client app requests for another PoP token with its own kid, we want to be able to request new tokens for the new kid instead of returning the previous PoP token (because the AT credential filter would be a match in the cache). The externalKeyId helps make that differentiation.

Reply:
Addressed based on our discussion.